src/main/java/com/sprint/mission/discodeit/security/jwt/JwtService.java

```diff
@@ -0,0 +1,177 @@
+    User user = userRepository.findById(userDto.id())
+        .orElseThrow(() -> new RuntimeException("User not found"));
+
+    JwtSession jwtSession = new JwtSession();
+    jwtSession.setUser(user);
+    jwtSession.setRefreshToken(refreshToken);
+    jwtSession.setRevoked(false);
+    jwtSession.setExpiresAt(LocalDateTime.now().plus(refreshTokenValidity));
+    jwtSessionRepository.save(jwtSession);
+  }
+
+  public boolean isValidRefreshToken(String token) {
+    try {
+      Claims claims = parseClaims(token);
+      return "refreshToken".equals(claims.getSubject());
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+
+  public boolean isValidAccessToken(String token) {
+    try {
+      Claims claims = parseClaims(token);
+      return claims.getSubject().equals("accessToken");
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public Claims parseClaims(String token) {
+    return Jwts.parserBuilder()
+        .setSigningKey(getSecretKey())
+        .build()
+        .parseClaimsJws(token)
+        .getBody();
+  }
+
+
+  @Transactional
+  public String refreshAccessToken(String refreshToken) {
+    JwtSession session = getSessionByRefresh(refreshToken)
+        .orElseThrow(InvalidRefreshTokenException::expiredOrInvalid);
+
+    UserDto userDto = userMapper.toDto(session.getUser());
+    return generateAccessToken(userDto);
+  }
+
+
+  public Optional<JwtSession> getSessionByRefresh(String refreshToken) {
+    return jwtSessionRepository.findByRefreshTokenAndRevokedFalse(refreshToken);
+  }
+
+
+  public void revokeRefreshToken(String refreshToken) {
+    Optional<JwtSession> session = jwtSessionRepository.findByRefreshToken(refreshToken);
+    session.ifPresent(jwtSession -> {
+      jwtSession.setRevoked(true);
+      jwtSessionRepository.save(jwtSession);
+    });
+  }
+
+  @Transactional
+  public void forceLogoutByUserId(UUID userId) {
+    jwtSessionRepository.deleteAllByUserId(userId);
+  }
+
+  public void removeTokenCookies(HttpServletResponse response) {
+    deleteCookie(response, "refreshToken");
+    deleteCookie(response, "accessToken");
+  }
+
+  private void deleteCookie(HttpServletResponse response, String name) {
+    ResponseCookie cookie = ResponseCookie.from(name, null)
+        .httpOnly(false)
+        .secure(false)
+        .path("/")
+        .maxAge(Duration.ZERO)
+        .build();
+    response.addHeader("Set-Cookie", cookie.toString());
+  }
+  public void handleLoginSuccess(UserDto userDto, HttpServletResponse response) throws IOException {
+    String accessToken = generateAccessToken(userDto);
+    String refreshToken = generateRefreshToken(userDto);
+    saveJwtSession(userDto, refreshToken);
+
+    ResponseCookie refreshCookie = ResponseCookie.from("refreshToken", refreshToken)
+        .httpOnly(true)
+        .secure(false)
+        .path("/")
+        .maxAge(refreshTokenValidity)
+        .build();
+    response.addHeader("Set-Cookie", refreshCookie.toString());
```

Comment:
```suggestion
    response.addHeader(HttpHeaders.SET_COOKIE, refreshCookie.toString());
```
상수가 정의되어 있어요. 위 방식은 어때요?

그리고 ResponseCookie 잘 활용해주셨네요!👍